### PR TITLE
Add chart to DAO dashboard

### PR DESCRIPTION
**Pull request by rubycop**
Screenshot 2024-12-15 at 17 18 28

### CODE REVIEW

**Comment by rubycop** [comment]
@race-of-sloths

**Comment by rubycop** [comment]
@race-of-sloths include

**Comment by race-of-sloths** [comment]
@rubycop Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!


[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://race-of-sloths.com/images/bot-banner-dark.svg">
    <source media="(prefers-color-scheme: light)" srcset="https://race-of-sloths.com/images/bot-banner-light.svg">
    Shows inviting banner with latest news.
</picture>](https://race-of-sloths.com/latest-news)

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/rubycop?type=bot&pr=NEAR-DevHub/neardevhub-treasury-dashboard/152&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/rubycop?type=bot&pr=NEAR-DevHub/neardevhub-treasury-dashboard/152&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/rubycop)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command. Alternatively, autoscoring [1,2] will be applied for this pull request

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/rubycop)

For maintainers:
- Score pull requests that participate in the Race of Sloths and receive a reward
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Comment by petersalomonsen** [comment]
@rubycop would it be possible to add a playwright test that shows all the usage scenarios? also would be great then to run the test with video mode on, and upload the video files here.

File: instances/treasury-devdao.near/widget/pages/dashboard/Portfolio.jsx
Comment (Megha-Dev-19) [inline instances/treasury-devdao.near/widget/pages/dashboard/Portfolio.jsx:141]: To stay consistent with the decimal places
```suggestion
                ${Big(price ?? "0").toFixed(2)}
```

**Review by Megha-Dev-19 — APPROVED** [review]
LGTM, would be great to have tests. Pre-approving

**Comment by rubycop** [comment]
@petersalomonsen 
test scenario:
1. chart is loading
2. mouse move is working and changes data
3. switching date period changes data on chart
https://github.com/user-attachments/assets/7fd58907-f6fb-4934-8630-fbc11d1f7eb1

**Comment by petersalomonsen** [comment]
@race-of-sloths score 5

File: playwright-tests/tests/dashboard.treasury-devdao.near.spec.js
Comment (petersalomonsen) [inline playwright-tests/tests/dashboard.treasury-devdao.near.spec.js:21]: Can you insert an assertion here on what timestamp that should be shown when hovering the mouse?

Also I am wondering why the y-value is changing in the chart when moving the mouse?

File: playwright-tests/tests/dashboard.treasury-devdao.near.spec.js
Comment (petersalomonsen) [inline playwright-tests/tests/dashboard.treasury-devdao.near.spec.js:24]: Please insert an assertion that also the time labels changes when changing the time resolution on the charts

**Comment by petersalomonsen** [comment]
Note that you can record test videos by using the `video` property in [playwright.config.js](https://github.com/NEAR-DevHub/neardevhub-treasury-dashboard/blob/main/playwright.config.js#L37)

File: playwright-tests/tests/dashboard.treasury-devdao.near.spec.js
Comment (rubycop) [reply playwright-tests/tests/dashboard.treasury-devdao.near.spec.js:21]: it's started animation when chart appears - it shows up a line liike in video

**Pull request merged**
